onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorSlice.cpp

```diff
@@ -6,16 +6,26 @@
 namespace Dml
 {
 
-class DmlOperatorSlice : public DmlOperator, public SliceHelper
+template <uint32_t opsetVersion>
```

Comment:
Let's avoid unnecessary template bloat by passing this as a constructor parameter. See DmlOperatorSlice.cpp for an example (the 5 lines of indirection via `DmlOperatorPoolingTemplate` saves 80 lines).